mizar/daemon/interface_service.py

```diff
@@ -219,10 +227,23 @@ def ConsumeInterfaces(self, request, context):
         logger.error("Timeout, no new interface to consume!")
         return self._ConsumeInterfaces(requested_pod_name, cni_params)
 
-    def DeleteInterface(self, request, context):
-        # TODO (Cathy): implement the reverse logic
-        return empty_pb2.Empty()
+    def DeleteInterfaces(self, request, context):
+        """
+        Delete network interfaces for a pod
+        """
+        # input is pod_id 
 
+        cni_params = request
+        pod_name = get_pod_name(cni_params.pod_id)
+        interfaces = self.interfaces.get(pod_name, [])
+        logger.debug("Deleting interfaces for pod {}".format(pod_name))
+        print(interfaces)
+        for interface in interfaces:
+            if pod_name in self.interfaces:
+                self.interfaces[pod_name].remove(interface)
+            self._DeleteInterfaces(pod_name, interface.interface_id)
```

Comment:
The function signatutre takes: pod_name, cni_paramter. Here it passed interface_id as the second parameter. Then the function itself does not make any use of the second parameter. I believe the _DeleteInterfaces function must delete only the interface that the CNI requests. Would you please fix this?